pkg/registry/core/componentstatus/rest.go

```diff
@@ -96,7 +93,7 @@ func ToConditionStatus(s probe.Result) api.ConditionStatus {
 }
 
 func (rs *REST) getComponentStatus(name string, server Server) *api.ComponentStatus {
-	status, msg, err := server.DoServerCheck(rs.prober)
+	status, msg, err := server.DoServerCheck()
```

Comment:
Since you take `server` by value, it is going to have to construct a prober every time, it won't cache it. I guess you probably want to start storing pointers in the map.